splitmatrix.cpp

```diff
@@ -165,16 +170,19 @@ int SplitMatrix::createDistanceFilesFromTax(map<string, int>& seqGroup, int numG
 				if ((namefile == "") && (countfile == "")) {  names.insert(query.getName()); }
 			
 				if (it != seqGroup.end()) { //not singleton 
-					m->openOutputFileAppend((fastafile + "." + toString(it->second) + ".temp"), outFile);
-					query.printSequence(outFile); 
-					outFile.close();
-					
+					query.printSequence(*outFiles[it->second]);
 					copyGroups.erase(query.getName());
 				}
 			}
 		}
 		in.close();
-		
+
+		//Close output files
+		for (map<int, ofstream*>::iterator it = outFiles.begin(); it != outFiles.end(); it++) {
+			delete it->second;
```

Comment:
Thanks for the code contribution! Can you add a line to close the file before the delete?